### PR TITLE
PojavLauncher Updates

### PR DESCRIPTION
**Pull request by Doregon**
We’ve added many new features and visual enhancements to PojavLauncher, the main changes being that there is now a LaunchScreen.storyboard and Assets.xcassets to compile. I had a draft pull request already open for this, but with the move to the main branch I managed to vaporize it by pulling from upstream.

Just like the last one, this is not yet mergeable. Once we get a new version released, or when we ask you to update, then this needs to be merged to be able to build.

### CODE REVIEW

**Review by asdfugil — CHANGES_REQUESTED** [review]
change this to `ifneq (,$(findstring iphoneos,$(MEMO_TARGET)))`

**Comment by Doregon** [comment]
Lol I have no idea what happened there

File: makefiles/pojavlauncher.mk [hunk 1]
```diff
@@ -75,4 +77,4 @@ pojavlauncher-package: pojavlauncher-stage
 
 endif # ($(filter $(shell uname -m | cut -c -4), iPad iPho),)
 endif # ($(UNAME),Darwin)
-endif # ($(MEMO_TARGET),iphoneos-arm64)
\ No newline at end of file
+endif # ($(MEMO_TARGET),iphoneos-arm64)
```
Comment (asdfugil) [inline makefiles/pojavlauncher.mk:80]: Please also change this comment

Comment (Doregon) [reply makefiles/pojavlauncher.mk]: bonk

**Review by asdfugil — APPROVED** [review]

File: makefiles/pojavlauncher.mk [hunk 2]
```diff
@@ -33,10 +33,12 @@ pojavlauncher: pojavlauncher-setup
...
 			cd build; \
 			wget https://github.com/leetal/ios-cmake/raw/master/ios.toolchain.cmake; \
-			cmake .. -G Xcode -DDEPLOYMENT_TARGET=$(IPHONEOS_DEPLOYMENT_TARGET) -DCMAKE_TOOLCHAIN_FILE=ios.toolchain.cmake -DPLATFORM=OS64 -DENABLE_BITCODE=FALSE -DCMAKE_XCODE_ATTRIBUTE_CODE_SIGNING_REQUIRED="NO" -DCMAKE_XCODE_ATTRIBUTE_CODE_SIGNING_ALLOWED=NO -DCMAKE_XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY=""; \
+			cmake .. -G Xcode -DCMAKE_TOOLCHAIN_FILE=ios.toolchain.cmake -DDEPLOYMENT_TARGET=$(IPHONEOS_DEPLOYMENT_TARGET) -DENABLE_ARC=TRUE -DENABLE_VISIBILITY=FALSE -DPLATFORM=OS64 -DENABLE_BITCODE=FALSE -DENABLE_STRICT_TRY_COMPILE=FALSE -DCMAKE_XCODE_ATTRIBUTE_CODE_SIGNING_REQUIRED="NO" -DCMAKE_XCODE_ATTRIBUTE_CODE_SIGNING_ALLOWED=NO -DCMAKE_XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY=""; \
 			cmake --build . --config Release --target pojavexec PojavLauncher; \
 			cd ../..; \
 			mkdir -p Natives/build/Release-iphoneos/PojavLauncher.app/Base.lproj; \
+			/Applications/Xcode.app/Contents/Developer/usr/bin/actool Natives/Assets.xcassets --compile Natives/resources --platform iphoneos --minimum-deployment-target $(IPHONEOS_DEPLOYMENT_TARGET) --app-icon AppIcon --output-partial-info-plist /dev/null; \
```
Comment (Diatrus) [inline makefiles/pojavlauncher.mk:40]: Can you change this to just “xcrun actool”

**Review by Diatrus — CHANGES_REQUESTED** [review]

File: makefiles/pojavlauncher.mk (hunk 2 above)
Comment (Doregon) [reply makefiles/pojavlauncher.mk:40]: If I had the fr emote available here, I would’ve used it. In my other pull that vanished, you said to use the direct path.

Comment (nyuszika7h) [reply makefiles/pojavlauncher.mk]: fr

Comment (asdfugil) [reply makefiles/pojavlauncher.mk]: fr chain
fr

Comment (Doregon) [reply makefiles/pojavlauncher.mk]: !fr

Comment (asdfugil) [reply makefiles/pojavlauncher.mk]: oof

Comment (Doregon) [reply makefiles/pojavlauncher.mk]: fr

**Comment by Doregon** [comment]
I just updated the makefile to match somewhat with our new one, most notably the lack of Gradle and the unified CMake script.

**Comment by Diatrus** [comment]
Please don’t have a JAVAFILES var. have a POJAV_JAVAFILES.

File: makefiles/pojavlauncher.mk [hunk 3]
```diff
@@ -26,29 +28,33 @@ pojavlauncher:
...
-	# Reimplement the build script
+	# Copy and paste from the new Makefile... basically
 	cd $(BUILD_WORK)/pojavlauncher; \
 		unset CC CXX LD CFLAGS CPPFLAGS CXXFLAGS LDFLAGS; \
 		cd Natives; \
 			mkdir -p build; \
 			cd build; \
-			wget https://github.com/leetal/ios-cmake/raw/master/ios.toolchain.cmake; \
-			cmake .. -G Xcode -DDEPLOYMENT_TARGET=$(IPHONEOS_DEPLOYMENT_TARGET) -DCMAKE_TOOLCHAIN_FILE=ios.toolchain.cmake -DPLATFORM=OS64 -DENABLE_BITCODE=FALSE -DCMAKE_XCODE_ATTRIBUTE_CODE_SIGNING_REQUIRED="NO" -DCMAKE_XCODE_ATTRIBUTE_CODE_SIGNING_ALLOWED=NO -DCMAKE_XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY=""; \
+			cmake . -DCMAKE_BUILD_TYPE=Release -DCMAKE_CROSSCOMPILING=true -DCMAKE_SYSTEM_NAME=Darwin -DCMAKE_SYSTEM_PROCESSOR=aarch64 -DCMAKE_OSX_SYSROOT="$(SDKPATH)" -DCMAKE_OSX_ARCHITECTURES=arm64 -DCMAKE_C_FLAGS="-arch arm64 -miphoneos-version-min=$(IPHONEOS_DEPLOYMENT_TARGET)" ..
```
Comment (Diatrus) [inline makefiles/pojavlauncher.mk:37]: DEFAULT_CMAKE_FLAGS?

File: makefiles/pojavlauncher.mk [hunk 4]
```diff
@@ -26,29 +28,33 @@ pojavlauncher:
...
 			cmake --build . --config Release --target pojavexec PojavLauncher; \
 			cd ../..; \
-			mkdir -p Natives/build/Release-iphoneos/PojavLauncher.app/Base.lproj; \
-			ibtool --compile Natives/build/Release-iphoneos/PojavLauncher.app/Base.lproj/MinecraftSurface.storyboardc Natives/en.lproj/MinecraftSurface.storyboard; \
-			mkdir -p Natives/build/Release-iphoneos/PojavLauncher.app/Frameworks; \
-			cp Natives/build/Release-iphoneos/libpojavexec.dylib Natives/build/Release-iphoneos/PojavLauncher.app/Frameworks/; \
-			cp -R Natives/resources/* Natives/build/Release-iphoneos/PojavLauncher.app/; \
+			mkdir -p Natives/build/PojavLauncher.app/Base.lproj; \
+			actool Natives/Assets.xcassets --compile Natives/resources --platform iphoneos --minimum-deployment-target $(IPHONEOS_DEPLOYMENT_TARGET) --app-icon AppIcon --output-partial-info-plist /dev/null; \
```
Comment (Diatrus) [inline makefiles/pojavlauncher.mk:41]: There is a platform var you should use here

File: makefiles/pojavlauncher.mk (hunk 3 above)
Comment (Doregon) [reply makefiles/pojavlauncher.mk]: done

File: makefiles/pojavlauncher.mk (hunk 4 above)
Comment (Doregon) [reply makefiles/pojavlauncher.mk]: done

**Comment by Doregon** [comment]
> Please don’t have a JAVAFILES var. have a POJAV_JAVAFILES.

also done :)

**Comment by Doregon** [comment]
Alright time to finish this pull request, however before I can I need to get OpenJDK 8 on Procursus so your build machine is able to produce a working copy. I'm going to close this and refork Procursus so I'm on a clean slate, this may be a lot of work.

**Pull request closed without merging**